### PR TITLE
Writing output artifacts of generated assignments

### PR DESCRIPTION
**Pull request by aleasims**
This has to be merged after this PR in blueprint:

- NilFoundation/zkllvm-blueprint/pull/384

Hopefully this code is going to be removed soon (and replaced with proper assignment tables tooling), so I wasn't very precise about it.

### CODE REVIEW

File: libs/output_artifacts/src/output_artifacts.cpp [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    }
+    return res;
+}
+
+std::expected<Range::ConcreteRange, std::string> Range::concrete_range(int max) const {
+    int upper_ = upper.value_or(max);
+    if (upper_ > max || upper_ < 0 || lower > upper_) {
+        return std::unexpected("index out of bounds");
+    }
+    return std::make_pair(lower, upper_);
```
[the remark this answers is not in the document]
Comment (akokoshn) [reply libs/output_artifacts/src/output_artifacts.cpp:66]: Have you seen any warnings?

[the remark this answers is not in the document]
Comment (aleasims) [reply libs/output_artifacts/src/output_artifacts.cpp:66]: Yep, but `upper_` is checked here to be non-negative, so it's fine. I can add explicit conversion for clarity I you want.

[the remark this answers is not in the document]
Comment (akokoshn) [reply libs/output_artifacts/src/output_artifacts.cpp:66]: yes, thanks! I just want to avoid warnings

[the remark this answers is not in the document]
Comment (aleasims) [reply libs/output_artifacts/src/output_artifacts.cpp:66]: There are no warnings for that with current compilation flags

**Review by akokoshn — APPROVED** [review]

**Pull request merged**